scm/src/gmtb_scm_setup.f90

```diff
@@ -42,72 +42,131 @@ subroutine set_state(scm_input, scm_reference, scm_state)
+         end do
+       end if
+     end do
+   
+     !> - For each column, interpolate the ozone to the model grid.
+     if(scm_state%ozone_index > 0) then
+       do i=1, scm_state%n_cols
+         call interpolate_to_grid_centers(scm_input%input_nlev, scm_input%input_pres, scm_input%input_ozone, scm_state%pres_l(i,1,:), &
+           scm_state%n_levels, scm_state%state_tracer(i,1,:,scm_state%ozone_index,1), last_index_init, 1)
+         !>  - If the input domain does not span the model domain, patch in McClatchey tropical standard atmosphere (smoothly over a number of levels) above.
+         if(last_index_init < scm_state%n_levels) THEN
+           call patch_in_ref(last_index_init, scm_state%n_levels_smooth, scm_reference%ref_nlev, scm_reference%ref_pres, &
+             scm_reference%ref_ozone, scm_state%pres_l(i,1,:), scm_state%n_levels, &
+             scm_state%state_tracer(i,1,:,scm_state%ozone_index,1), grid_error)
+         end if
+       end do
+     end if
+   
+     scm_state%state_tracer(:,1,:,scm_state%cloud_water_index,1) = 0.0
+   else
+     do i=1, scm_state%n_cols
+        !input_T = (scm_input%input_pres/p0)**con_rocp*(scm_input%input_thetail + (con_hvap/con_cp)*scm_input%input_ql + (con_hfus/con_cp)*scm_input%input_qi)
+        scm_state%state_u(i,1,:,1) = scm_input%input_u(:)
+        scm_state%state_v(i,1,:,1) = scm_input%input_v(:)
+        scm_state%state_T(i,1,:,1) = scm_input%input_temp(:)
+        scm_state%state_tracer(i,1,:,scm_state%water_vapor_index,1)=scm_input%input_qt
+        scm_state%state_tracer(i,1,:,scm_state%ozone_index,1)=scm_input%input_ozone
+        scm_state%veg_type = scm_input%input_vegtyp
+        scm_state%soil_type = scm_input%input_soiltyp
+        scm_state%veg_type_real = DBLE(scm_input%input_vegtyp)
+        scm_state%veg_frac_real = scm_input%input_vegfrac
+        scm_state%slopetype     = scm_input%input_slopetype
+        scm_state%shdmin = scm_input%input_shdmin  
+        scm_state%shdmax = scm_input%input_shdmax  
+        scm_state%sfc_roughness_length_cm = scm_input%input_zorl    
+        scm_state%sfc_type_real = scm_input%input_slmsk   
+        scm_state%canopy = scm_input%input_canopy  
+        scm_state%hice = scm_input%input_hice  
+        scm_state%fice = scm_input%input_fice  
+        scm_state%tisfc = scm_input%input_tisfc  
+        scm_state%snwdph = scm_input%input_snwdph  
+        scm_state%snoalb = scm_input%input_snoalb  
+        scm_state%sncovr = scm_input%input_sncovr  
+        scm_state%area = scm_input%input_area    
+        scm_state%tg3    = scm_input%input_tg3     
+        scm_state%uustar = scm_input%input_uustar  
+        scm_state%soil_type_real = DBLE(scm_input%input_soiltyp)
+        scm_state%stc(i,1,:,1)=scm_input%input_stc
+        scm_state%smc(i,1,:,1)=scm_input%input_smc
+        scm_state%slc(i,1,:,1)=scm_input%input_slc
+        if (scm_input%input_pres_i(1).GT. 0.0) then ! pressure are read in, overwrite values
```

Comment:
Is this check for input_pres_i > 0 really needed? Can we assume that if we have FV3GFS ICs, that pres_i, and pres_l will just be provided already?

Reply:
Actually, the extract_FV3GFS_column_ic.py script only pulls out surface pressure and the SCM uses the aks and bks to calculate pres_i and pres_l.  I want this functionality there for more through debugging of a model crash. Since the model full model is non-hydrostatic, the aks and bks do not give the exact pressure after the initial time.  I have a different python script that can pull out more information from the history files of a run to diagnose a crash.

Reply:
OK, thanks for the explanation.